test/parallel/test-process-reportError.mjs

```diff
@@ -0,0 +1,11 @@
+import { mustCall, mustNotCall } from '../common/index.mjs';
+import { strictEqual } from 'assert';
+{
+  // Verify uncaughtException is called
+  process.on('unhandledRejection', mustNotCall());
+  const e = new Error('boom');
+  process.on('uncaughtException', mustCall((error) => {
+    strictEqual(error, e);
+  }));
+  reportError(e);
+}
```

Comment:
```suggestion

// Verify uncaughtException is called
process.on('unhandledRejection', mustNotCall());
const e = new Error('boom');
process.on('uncaughtException', mustCall((error) => {
  strictEqual(error, e);
}));
reportError(e);
```
nit

Reply:
I'll add more tests to justify the scoping :)